apps/examples/testcase/le_tc/kernel/itc_pthread.c

```diff
@@ -0,0 +1,522 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/// @file itc_pthread.c
+/// @brief ITC Test Case Example for Pthread API
+
+/****************************************************************************
+ * Included and defines
+ ****************************************************************************/
+
+#include <pthread.h>
+#include <stdio.h>
+#include <errno.h>
+#include <pthread.h>
+#include <stdlib.h>
+#include <tinyara/pthread.h>
+#include <tinyara/config.h>
+#include "tc_internal.h"
+#include <sys/types.h>
+#include <stdbool.h>
+#include <unistd.h>
+#include <signal.h>
+#include <sched.h>
+
+#define WAIT_TIME_1                   1
+#define WAIT_TIME_2                   2
+#define WAIT_TIME_4                   4
+#define WAIT_TIME_8                   8
+#define WAIT_TIME_0                   0
+#define BUF_LEN 32
+#define INIT_MINUSONE -1
+#define SIG_ERROR 0
+#define LOOP_SIZE CONFIG_ITC_KERN_PTHREAD_LOOP_SIZE
+
+static pthread_t g_pthread;
+static bool g_flag;
+static pthread_mutex_t g_mutex = PTHREAD_MUTEX_INITIALIZER;
+static pthread_key_t g_key;
+static int  g_exec_index;
+static bool g_bpthreadcallback = false;
+
+/****************************************************************************
+*Helper Functions
+**************************************************************************/
+/**
+* @fn                   :thread_setgetname
+* @brief                :utility function for creating thread
+* @return               :void*
+*/
+static void *thread_setgetname(void *param)
+{
+	while (1) {
+		sleep(WAIT_TIME_1);
+		if (g_exec_index == LOOP_SIZE) {
+			break;
+		}
+	}
+	return NULL;
+}
+/**
+ *@fn					:thread_setget
+ *@breif				:utility function for creating thread
+ *@return               :void*
+ */
+static void *thread_setget(void *param)
+{
+	sleep(WAIT_TIME_2);
+	return NULL;
+}
+
+/**
+* @fn                   :thread_equal_check
+* @brief                :utility function for itc_pthread_pthread_equal_p
+* @return               :void*
+*/
+static void *thread_equal_check(void *param)
+{
+	if (pthread_equal(g_pthread, pthread_self()) == false) {
+		g_flag = false;
+		return NULL;
+	}
+	g_flag = true;
+	return NULL;
+}
+
+/**
+* @fn                   :thread_set_get_specific
+* @brief                :utility function for itc_pthread_key_create_set_getspecific_p_multitime
+* @return               :void*
+*/
+static void *thread_set_get_specific(void *param)
+{
+	int *get_spec;
+	int ret_chk;
+	int exec_index;
+	g_flag = false;
+
+	for (exec_index = 1; exec_index <= LOOP_SIZE; exec_index++) {
+		ret_chk = pthread_setspecific(g_key, &exec_index);
+		if (ret_chk != OK) {
+			printf("itc_pthread_key_create_set_getspecific_p_multitime: TC FAIL\n");
```

Comment:
This printed msg is not matched our TC output format

Reply:
corrected printf format as per utc code.